src/contracts/MsVault.h

```diff
@@ -0,0 +1,708 @@
+    };
+
+    struct isOwnerOfVault_input
+    {
+        Vault vault;
+        id ownerID;
+    };
+    struct isOwnerOfVault_output
+    {
+        bit result;
+    };
+    struct isOwnerOfVault_locals
+    {
+        findOwnerIndexInVault_input fi_in;
+        findOwnerIndexInVault_output fi_out;
+        findOwnerIndexInVault_locals fi_locals;
+    };
+
+    struct resetReleaseRequests_input
+    {
+        Vault vault;
+    };
+    struct resetReleaseRequests_output
+    {
+        Vault vault;
+    };
+    struct resetReleaseRequests_locals
+    {
+        sint32 i;
+    };
+
+    struct registerVault_input
+    {
+        uint16 vaultType;
+        id vaultName;
+        array<id, MAX_OWNERS> owners;
+    };
+    struct registerVault_output
+    {
+    };
+    struct registerVault_locals
+    {
+        uint16 ownerCount;
+        uint16 i, j;
+        sint32 slotIndex;
+        Vault newVault;
+
+        resetReleaseRequests_input rr_in;
+        resetReleaseRequests_output rr_out;
+        resetReleaseRequests_locals rr_locals;
+    };
+
+    struct deposit_input
+    {
+        uint64 vaultID;
+    };
+    struct deposit_output
+    {
+    };
+    struct deposit_locals
+    {
+        Vault vault;
+    };
+
+    struct releaseTo_input
+    {
+        uint64 vaultID;
+        uint64 amount;
+        id destination;
+    };
+    struct releaseTo_output
+    {
+    };
+    struct releaseTo_locals
+    {
+        Vault vault;
+        MSVaultLogger logger;
+
+        sint32 ownerIndex;
+        uint16 approvals;
+        uint16 totalOwners;
+        bit releaseApproved;
+        uint16 requiredApprovals;
+        uint16 i;
+
+        isOwnerOfVault_input io_in;
+        isOwnerOfVault_output io_out;
+        isOwnerOfVault_locals io_locals;
+
+        findOwnerIndexInVault_input fi_in;
+        findOwnerIndexInVault_output fi_out;
+        findOwnerIndexInVault_locals fi_locals;
+
+        resetReleaseRequests_input rr_in;
+        resetReleaseRequests_output rr_out;
+        resetReleaseRequests_locals rr_locals;
+    };
+
+    struct resetRelease_input
+    {
+        uint64 vaultID;
+    };
+    struct resetRelease_output
+    {
+    };
+    struct resetRelease_locals
+    {
+        Vault vault;
+        MSVaultLogger logger;
+        sint32 ownerIndex;
+
+        isOwnerOfVault_input io_in;
+        isOwnerOfVault_output io_out;
+        isOwnerOfVault_locals io_locals;
+
+        findOwnerIndexInVault_input fi_in;
+        findOwnerIndexInVault_output fi_out;
+        findOwnerIndexInVault_locals fi_locals;
+
+        bit found;
+    };
+
+    struct getVaults_input
+    {
+        id publicKey;
+    };
+    struct getVaults_output
+    {
+        uint16 numberOfVaults;
+        array<uint64, 1024> vaultIDs;
+        array<id, 1024> vaultNames;
+    };
+    struct getVaults_locals
+    {
+        uint16 count;
+        uint16 i, j;
+        Vault v;
+    };
+
+    struct getReleaseStatus_input
+    {
+        uint64 vaultID;
```

Comment:
the term vaultID is used at several places.
internally it is the the index, one could think about name it also index to make it also clear for anyone which is using the SC.

if we keep id, i personally would also write `vaultId` instead of `vaultID` to keep it all in camal case